client/landing/jetpack-cloud/components/scan-threats/index.tsx

```diff
@@ -44,27 +45,52 @@ const ScanThreats = ( { site, threats }: Props ) => {
 	const dispatch = useDispatch();
 
 	const openFixAllThreatsDialog = React.useCallback( () => {
+		dispatch(
+			recordTracksEvent( `calypso_scan_all_threats_dialog_open`, {
+				site_id: site.ID,
+			} )
+		);
 		setShowFixAllThreatsDialog( true );
-	}, [] );
+	}, [ dispatch, site ] );
 
-	const openDialog = React.useCallback( ( action: ThreatAction, threat: Threat ) => {
-		setSelectedThreat( threat );
-		setActionToPerform( action );
-		setShowThreatDialog( true );
-	}, [] );
+	const openDialog = React.useCallback(
+		( action: ThreatAction, threat: Threat ) => {
+			dispatch(
+				recordTracksEvent( `calypso_scan_${ action }_threat_dialog_open`, {
+					site_id: site.ID,
+					threat_id: threat.id,
+				} )
+			);
+			setSelectedThreat( threat );
+			setActionToPerform( action );
+			setShowThreatDialog( true );
+		},
+		[ dispatch, site ]
+	);
 
 	const closeDialog = React.useCallback( () => {
 		setShowThreatDialog( false );
 	}, [] );
 
 	const confirmAction = React.useCallback( () => {
+		dispatch(
+			recordTracksEvent( `calypso_scan_threat_${ actionToPerform }`, {
```

Comment:
Lets leave a comment here so that we can find the different possible event names in the future.

Reply:
Yes. After writing this part, I read our internal guide on how to write good event names. The use of variables inside strings is not recommended so I'll make those strings or leave a comment.